app/forms/idv/api_image_upload_form.rb

```diff
@@ -49,15 +43,15 @@ def liveness_checking_enabled?
     end
 
     def front
-      params[:front]
+      as_readable(params[:front])
     end
 
     def back
-      params[:back]
+      as_readable(params[:back])
     end
 
     def selfie
-      params[:selfie]
+      as_readable(params[:selfie])
```

Comment:
this is going to memoize based on the whole value, which could be big/medium, what if we memoized based on the key?

```suggestion
      as_readable(:selfie)
```

Reply:
> this is going to memoize based on the whole value, which could be big/medium, what if we memoized based on the key?

I like that 👍  Also, it occurs to me that `params[:selfie]` may be _either_ a string or a `ActionDispatch::Http::UploadedFile` which... if my understanding of Ruby is correct, would not be a valid key 😬

Reply:
> this is going to memoize based on the whole value, which could be big/medium, what if we memoized based on the key?

Updated in d7e26429f.

Reply:
Hashing is fine --- all objects in Ruby have `#hash` which returns an int, similar to Java

Reply:
changes LG tho! 👍